src/Cli/Utils.cs

```diff
@@ -669,11 +670,20 @@ public static bool TryGetRoleAndOperationFromPermission(IEnumerable<string> perm
             }
             else
             {
+                // For Views and Tables
                 if (parameters is not null && parameters.Any())
                 {
                     _logger.LogError("Tables/Views don't support parameters.");
                     return false;
                 }
+
+                // For Views
+                if (sourceType is SourceType.View && (keyFields is null || !keyFields.Any()))
```

Comment:
Even though this change is scoped to CLI to prevent breaking changes in the engine, it would be helpful to ensure (and add if missing) a test that validates the engine doesnt crash when supplied a view entity with missing key fields , especially for pgsql. So, if we ever decide to enforce this change in the engine, we will have to fix this test. The test would also prevent an accidental regression if we accidentally made this breaking change before being ready.

Reply:
Actually, engine will still crash with `Primary key not configured on the given database object {tableName}` when key-fields are not specified for pg-sql. We should have a test for that. 

For certain views in MsSql, key-fields are optional since they can still be inferred - we need a test to ensure such views continue to be supported for MsSql - even though addition of such view through CLI would be resulting in validation failures.

Reply:
Added a Test for MsSQL, where in the config the views with no keyfields are supported.